toolchain/check/testdata/facet/early_rewrites.carbon

```diff
@@ -80,16 +223,25 @@ interface Tuple {}
 impl () as Tuple {}
 class C(T:! Tuple) {}
 
-// TODO: Should this pass? Or do we need a `where` between .Z2 and .Z3 to make
-// .Z2's value visible to .Z3?
-// See https://github.com/carbon-language/carbon-lang/issues/5884.
 interface J {
-  // CHECK:STDERR: fail_todo_resolved_constraint_visible_in_type_parameter.carbon:[[@LINE+7]]:80: error: cannot convert type `.(Z.Z2)` into type implementing `Tuple` [ConversionFailureTypeToFacet]
-  // CHECK:STDERR:   let J1:! (Z & Y where .Y1 = ()) where .Z1 = .Self and .Z2 = .Z1.Y1 and .Z3 = C(.Z2);
-  // CHECK:STDERR:                                                                                ^~~~~~
-  // CHECK:STDERR: fail_todo_resolved_constraint_visible_in_type_parameter.carbon:[[@LINE-9]]:9: note: initializing generic parameter `T` declared here [InitializingGenericParam]
-  // CHECK:STDERR: class C(T:! Tuple) {}
-  // CHECK:STDERR:         ^
+  // Implied: .Y1 == (), but this is also explicitly required for J1.
+  // CHECK:STDERR: fail_todo_resolved_constraint_visible_in_type_parameter.carbon:[[@LINE+8]]:33: error: `let` declarations must end with a `;` [ExpectedDeclSemi]
+  // CHECK:STDERR:   let J1:! Z & Y where .Y1 = () where .Z1 = .Self and .Z2 = .Z1.Y1 and .Z3 = C(.Z2);
+  // CHECK:STDERR:                                 ^~~~~
+  // CHECK:STDERR:
+  // CHECK:STDERR: fail_todo_resolved_constraint_visible_in_type_parameter.carbon:[[@LINE+4]]:3: error: semantics TODO: `handle invalid parse trees in `check`` [SemanticsTodo]
+  // CHECK:STDERR:   let J1:! Z & Y where .Y1 = () where .Z1 = .Self and .Z2 = .Z1.Y1 and .Z3 = C(.Z2);
+  // CHECK:STDERR:   ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   // CHECK:STDERR:
-  let J1:! (Z & Y where .Y1 = ()) where .Z1 = .Self and .Z2 = .Z1.Y1 and .Z3 = C(.Z2);
+  let J1:! Z & Y where .Y1 = () where .Z1 = .Self and .Z2 = .Z1.Y1 and .Z3 = C(.Z2);
```

Comment:
This is failing with a syntactic error due to missing parens.
```suggestion
  let J1:! (Z & Y where .Y1 = ()) where .Z1 = .Self and .Z2 = .Z1.Y1 and .Z3 = C(.Z2);
```

Reply:
Ah, sorry I meant to simplify it by removing the nesting of `where` but left one in. It now fails with this error too: https://carbon.godbolt.org/z/xqrEG6nbc

So I split off a minimal example of that failure into another test as well.